pkg/controllers/securitypolicy/securitypolicy_controller.go

```diff
@@ -230,12 +228,6 @@ func getExistingConditionOfType(conditionType v1alpha1.ConditionType, existingCo
 func (r *SecurityPolicyReconciler) setupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.SecurityPolicy{}).
-		WithEventFilter(predicate.Funcs{
```

Comment:
when just deleting pods and leaving security policy behind, it would make legacy security policy which we should delete it. 
to my understanding, if user only delete pods instead of security policy, we just need to update effective members of security policy groups and should not delete it.

Reply:
> [1b3f3ef](https://github.com/vmware-tanzu/nsx-operator/commit/1b3f3ef4e8ad07186c9dc4e7b032370d64d1c22b)
> 
> we add this filter in this commit to avoid two times delete operation, if removing this one, we sill have the former issue?

Yes, it would report not finding sp error, no big deal though.

Reply:
> when just deleting pods and leaving security policy behind, it would make legacy security policy which we should delete it. to my understanding, if user only delete pods instead of security policy, we just need to update effective members of security policy groups and should not delete it.

It makes sense.